Makefile

```diff
@@ -622,19 +621,19 @@ kind-install-cilium-fast: kind-ready ## Install a local Cilium version into the
 
 .PHONY: build-cli
 build-cli: ## Build cilium cli binary
-	$(QUIET)$(MAKE) -C cilium-dbg
+	$(QUIET)$(MAKE) -C cilium-dbg GOOS=linux
```

Comment:
nit: I wonder whether it's worth doing a `GOOS ?= linux` definition somewhere like `Makefile.defs` so that this can be easily overridden as well. I have from time to time passed GOOS=darwin to try some things out, though I recognize that Cilium does not fully compile like that currently.

Reply:
Yeah I considered it, I wasn't sure about if I should do that or not. 

A global variable definition would change every invocation of our builds, whereas this is scoped to just the targets being used by the `kind-image-*-fast` targets, and as far as I can tell, these `build-*` targets are only used by the `kind-image-*-fast` targets. 

For example, if we blindly set `GOOS=linux` then running the tests will fail for non-Linux users, since it'll build the test binary for Linux, which cannot be executed on non-Linux.

Reply:
Ah good point, hmm. Alright I can't think of a better solution for now. We can always iterate on this in future. :+1: